src/vs/workbench/browser/parts/statusbar/statusbarPart.ts

```diff
@@ -502,10 +502,11 @@ export class StatusbarPart extends Part implements IStatusbarService {
 		const container = assertIsDefined(this.getContainer());
 		const styleOverride: IStatusbarStyleOverride | undefined = [...this.styleOverrides].sort((a, b) => a.priority - b.priority)[0];
 
-		// Background colors
+		// Background / foreground colors
 		const backgroundColor = this.getColor(styleOverride?.background ?? (this.contextService.getWorkbenchState() !== WorkbenchState.EMPTY ? STATUS_BAR_BACKGROUND : STATUS_BAR_NO_FOLDER_BACKGROUND)) || '';
 		container.style.backgroundColor = backgroundColor;
-		container.style.color = this.getColor(styleOverride?.foreground ?? (this.contextService.getWorkbenchState() !== WorkbenchState.EMPTY ? STATUS_BAR_FOREGROUND : STATUS_BAR_NO_FOLDER_FOREGROUND)) || '';
+		const foregroundColor = this.getColor(styleOverride?.foreground ?? (this.contextService.getWorkbenchState() !== WorkbenchState.EMPTY ? STATUS_BAR_FOREGROUND : STATUS_BAR_NO_FOLDER_FOREGROUND)) || '';
+		container.style.color = foregroundColor;
```

Comment:
The only problem with this part is that for themes that already define a focus border color + don't use a strong status bar bg color, they'll be impacted (like the GitHub Themes):

<img width="368" alt="CleanShot 2022-02-09 at 12 11 44@2x" src="https://user-images.githubusercontent.com/35271042/153282295-7f5a71e8-a221-4a82-8d48-6a7638c73409.png">


<img width="323" alt="CleanShot 2022-02-09 at 12 11 16@2x" src="https://user-images.githubusercontent.com/35271042/153282268-6c430ae3-0a82-4507-8758-161466d2202a.png">

So I think we'll likely need to create a new color token to allow themes to fine-tune this. If the majority of the most popular themes don't color their status bar then we should think about whether the default should just use the `focusBorder` as the default and then we customize it for our default themes.

Reply:
Good call. I'll see what I can do here

Reply:
I do not understand the concern. We will use these colors for the outline:
* `activeContrastBorder` if defined
* `styleOverride?.foreground` if defined
* `STATUS_BAR_FOREGROUND` if defined
* `STATUS_BAR_NO_FOLDER_FOREGROUND` if defined

If a theme disables `activeContrastBorder` then I think it is entirely not accessible. If a theme sets the other foreground colors to match the background color, the theme would be unusable because you could not read the status bar entries.

What am I missing?

Reply:
What about adding `statusBarItem.focusBorder` with a default value of `statusBar.foreground` that can be then set by theme authors? HC themes will still use `activeContrastBorder` hence being set to null below.

```
export const STATUS_BAR_ITEM_FOCUS_BORDER = registerColor('statusBarItem.focusBorder', {
	dark: STATUS_BAR_FOREGROUND,
	light: STATUS_BAR_FOREGROUND,
	hc: null,
}, localize('statusBarItemFocusBorder', "Status bar item border color when focused on keyboard navigation. The status bar is shown in the bottom of the window."));
```

Note the override for the GH themes to use their default `focusBorder`

https://user-images.githubusercontent.com/25163139/153482350-5f7deda6-4787-4ee2-af30-da549a827462.mp4

Reply:
That looks great to me, let's see if @bpasero has any concerns with that.

Reply:
Yeah agree, an explicit color for this makes good sense to me and is always a good approach to give theme authors most flexibility.

Reply:
Thanks both for the feedback! I'll get this merged in.